src/components/tables/FieldSelection/Rows.tsx

```diff
@@ -0,0 +1,116 @@
+import { TableCell, TableRow, Typography } from '@mui/material';
+import { CompositeProjection } from 'components/editor/Bindings/FieldSelection/types';
+import ChipListCell from 'components/tables/cells/ChipList';
+import ConstraintDetails from 'components/tables/cells/fieldSelection/ConstraintDetails';
+import FieldActions from 'components/tables/cells/fieldSelection/FieldActions';
+import { orderBy } from 'lodash';
+import { SortDirection } from 'types';
+
+interface RowProps {
+    row: CompositeProjection;
+}
+
+interface RowsProps {
+    data: CompositeProjection[];
+    sortDirection: SortDirection;
+    columnToSort: string;
+}
+
+function Row({ row }: RowProps) {
+    return (
+        <TableRow hover>
+            <TableCell>
+                <Typography>{row.field}</Typography>
+            </TableCell>
+
+            <TableCell>
+                <Typography>{row.ptr}</Typography>
+            </TableCell>
+
+            <ChipListCell values={row.inference.types} stripPath={false} />
+
+            {row.constraint ? (
+                <ConstraintDetails constraint={row.constraint} />
+            ) : null}
+
+            {row.constraint ? (
+                <FieldActions
+                    field={row.field}
+                    constraint={row.constraint}
+                    selectionType={row.selectionType}
+                />
+            ) : null}
```

Comment:
Unless the logic will change I think we could just make these two components get included with a single if.